datasets/dataset_cifar10.py

```diff
@@ -0,0 +1,78 @@
+# -*- coding: utf-8 -*-
+"""
+CIFAR10 dataset.
+"""
+
+from keras.datasets import cifar10
+from keras.models import Sequential
+from keras.layers import Dense, Dropout, Activation, Flatten
+from keras.layers import Conv2D, MaxPooling2D
+import keras
+
+
+# Data samples pre-processing method for inputs
+def preprocess_dataset_inputs(x):
+
+    x = x.astype("float32")
+    x /= 255
+
+    return x
+
+
+# Data samples pre-processing method for labels
+def preprocess_dataset_labels(y):
+
+    y = keras.utils.to_categorical(y, num_classes)
+
+    return y
+
+
+# Load and flatten data
+(x_train, y_train), (x_test, y_test) = cifar10.load_data()
+y_train = y_train.flatten()
+y_test = y_test.flatten()
+
+# Init dataset-specific variables
+input_shape = (32, 32, 3)
+num_classes = 10
+
+# Pre-process inputs
+x_train = preprocess_dataset_inputs(x_train)
+x_test = preprocess_dataset_inputs(x_test)
```

Comment:
I wonder why are the data are preprocessed here? It would be clearer to preprocess them  where the label are preprocessed

Reply:
Ah, I didn't see your comment

Reply:
Kind of a dilemma:
- I think the best would be that the user providing the dataset provides it all pre-processed. But thing is, for the labels it is not ideal, as we make quite some stuff with them during the data split. So the labels we have to put to binary vector after the data split
- So either we keep the input pre-processing in the datasets dedicated files, so that the user "delivers" clean inputs, or we put it after the data split but that entangles a bit more the dataset and the library logic

Reply:
ok we can let it like that then